landlab/components/space/tests/test_space.py

```diff
@@ -1,8 +1,234 @@
+        m_sp))), 1./n_sp)
+        
+    #test for match with analytical slope-area relationship
+    testing.assert_array_almost_equal(num_slope, analytical_slope, 
+                                      decimal=8, 
+                                      err_msg='SPACE transport-limited test failed',
+                                      verbose=True)
+    
+def test_matches_bedrock_alluvial_solution():
+    """
+    Test that model matches the bedrock-alluvial analytical solution
+    for slope/area relationship at steady state: 
+    S=((U * v_s * (1 - F_f)) / (K_sed * A^m) + U / (K_br * A^m))^(1/n).
+    
+    Also test that the soil depth everywhere matches the bedrock-alluvial
+    analytical solution at steady state: 
+    H = -H_star * ln(1 - (v_s / (K_sed / (K_br * (1 - F_f)) + v_s)))
+    """
+    
+    #set up a 3x3 grid with one open outlet node and low initial elevations.
+    nr = 5
+    nc = 5
+    mg = RasterModelGrid((nr, nc), 10.0)
+    
+    z = mg.add_zeros('node', 'topographic__elevation')
+    br = mg.add_zeros('node', 'bedrock__elevation')
+    soil = mg.add_zeros('node', 'soil__depth')
+
+    mg['node']['topographic__elevation'] += mg.node_y / 100000 \
+        + mg.node_x / 100000 \
+        + np.random.rand(len(mg.node_y)) / 10000
+    mg.set_closed_boundaries_at_grid_edges(bottom_is_closed=True,
+                                           left_is_closed=True,
+                                           right_is_closed=True,
+                                           top_is_closed=True)
+    mg.set_watershed_boundary_condition_outlet_id(0,
+                                                  mg['node']['topographic__elevation'], 
+                                                  -9999.)
+    soil[:] += 0. #initial soil depth of 100 m
+    br[:] = z[:]
+    z[:] += soil[:]
+    
+    #Instantiate DepressionFinderAndRouter
+    df = DepressionFinderAndRouter(mg)
+    
+    # Create a D8 flow handler
+    fa = FlowAccumulator(mg, flow_director='D8', 
+                         depression_finder='DepressionFinderAndRouter')
+    
+    # Parameter values for detachment-limited test
+    K_br = 0.02
+    K_sed = 0.02
+    U = 0.0001
+    dt = 1.0
+    F_f = 0.2 #all detached rock disappears; detachment-ltd end-member
+    m_sp = 0.5
+    n_sp = 1.0
+    v_s = 0.25
+    H_star = 0.1
+    
+    # Instantiate the Space component...
+    sp = Space(mg, K_sed=K_sed, K_br=K_br,
+                         F_f=F_f, phi=0.0, H_star=H_star, v_s=v_s,
+                         m_sp=m_sp, n_sp=n_sp, sp_crit_sed=0,
+                         sp_crit_br=0)
+
+    # ... and run it to steady state (4000x1-year timesteps).
+    for i in range(10000):
+        fa.run_one_step()
+        flooded = np.where(df.flood_status==3)[0]
+        sp.run_one_step(dt=dt, flooded_nodes=flooded)
+        br[mg.core_nodes] += U * dt #m
+        soil[0] = 0.
+        z[:] = br[:] + soil[:]
```

Comment:
similar to above, I don't totally understand this boundary condition. Why not enforced at the analytical steady state soil depth?

I see that you are comparing only the core nodes, but the slope downstream impacts qs-out (via Es and Er), so I'm having a hard time figuring out why it wouldn't matter.